src/components/App/index.jsx

```diff
@@ -42,8 +42,14 @@ const queryClient = new QueryClient({
 });
 
 const RedirectComponent = () => {
-  const location = useLocation();
-  return <Navigate to={`${location.pathname}/admin/${ROUTE_NAMES.learners}`} />;
+  const { enterpriseSlug } = useParams();
+
+  const homePage = generatePath(
+    `/:enterpriseSlug/admin/${ROUTE_NAMES.learners}`,
+    { enterpriseSlug },
+  );
```

Comment:
By directly referencing the enterprise slug from the route parameters, we are no longer reliant on the `pathname` from `useLocation`